man/systemd.unit.xml

```diff
@@ -1792,6 +1792,15 @@
           into.</para>
           </listitem>
         </varlistentry>
+
+        <varlistentry>
+            <term><varname>MaxInstances=</varname></term>
+
+            <listitem><para>The maximum number of instances of a template unit that may run simultaneously. If reached,
+            further attempts to start another instance of the same template will fail. Defaults to
+            <constant>unlimited</constant>.
```

Comment:
the parser doesn't actually accept the string unlimited, does it?

we should have an explicit parser for this i guess that can parse the word "infinity" and turns it into UINT_MAX.

(we use the word "infinity" already for unbounded timeouts, hence we should reuse this here I guess)

Reply:
and as mentioned I'd extend this concept to regular, non-templated units, where the only sane values would be 0 (refuse all start attempts) and 1 (allow them). If people specify > 1 this would not be any different than specifying 1, after all regular units only ever have one "instance" at max, so to speak